Scripts/Helpers/ProxyAffordanceMap.cs

```diff
@@ -0,0 +1,159 @@
+﻿#if UNITY_EDITOR
+using System;
+using System.Collections.Generic;
+using System.IO;
+using UnityEditor.Experimental.EditorVR.UI;
+using UnityEngine;
+using UnityEngine.InputNew;
+
+namespace UnityEditor.Experimental.EditorVR.Core
+{
+    [CreateAssetMenu(menuName = "EditorVR/EXR Proxy Affordance Map", fileName = "NewProxyAffordanceMap.asset")]
+    public class ProxyAffordanceMap : ScriptableObject
+    {
+        // TODO REMOVE - 5.6 HACK that remedies items not appearing in the create menu
+#if UNITY_EDITOR
+        [MenuItem("Assets/Create/EditorVR/EditorVR Proxy Affordance Map")]
+        public static void Create()
+        {
+            var path = AssetDatabase.GetAssetPath(Selection.activeObject);
+
+            if (string.IsNullOrEmpty(path))
+                path = "Assets";
+
+            if (!Directory.Exists(path))
+                path = Path.GetDirectoryName(path);
+
+            var affordanceMap = ScriptableObject.CreateInstance<ProxyAffordanceMap>();
+            path = AssetDatabase.GenerateUniqueAssetPath(path + "/NewProxyAffordanceMap.asset");
+            AssetDatabase.CreateAsset(affordanceMap, path);
+        }
+#endif
+
+        public enum VisibilityControlType
+        {
+            colorProperty,
+            alphaProperty,
+            materialSwap
+        }
+
+        [Serializable]
+        public class AffordanceVisibilityDefinition
+        {
+            [SerializeField]
+            VisibilityControlType m_VisibilityType;
+
+            [SerializeField] // colorProperty field
+            string m_ColorProperty = "_Color"; // Consider custom inspector that only displays this if this visibility type is chosen
+
+            [SerializeField] // alphaProperty field
+            string m_AlphaProperty = "_Alpha"; // Consider custom inspector that only displays this if this visibility type is chosen
```

Comment:
There isn't really a highlight state currently, just a normal & hidden state.  An affordance is only either visible or hidden currently.  I saw support for highlights as something we'd consider during the iconography pass, which could/would include additional arch to support displaying sprites "on" affordances/buttons... which will definitely require a group planning session before proceeding.  Due to potential shader variants that might/might-not support color changes for highlight, I feel that we should consider alternatives decoupled from shader driven approaches for the iconography pass.

Reply:
I think the visual definition of what something looks like should be captured in one place - not if that state is active, just what it looks like.  Not likely a deal-breaker for this PR - but if we imagine our users are going to add some pretty novel proxies and ways of visualizing them, it might also require pretty different methods of visualizing a highlight than our current methods.  I could certainly see that being the case for a rigged semi-transparent hand you see with something like the leapmotion.

Reply:
There is an opportunity for this in the implementation of the feedback receiver. If we want it to respond to feedback with different visuals, that is totally possible